### PR TITLE
feat(toolkit): add support for decoding data requests from .sol files

### PR DESCRIPTION
**Pull request by aesedepece**

### CODE REVIEW

File: toolkit/src/cli/data_requests.rs [hunk 1]
```diff
@@ -27,8 +36,29 @@ pub(crate) fn decode_from_args(
...
+/// that is an instance of the `WitnetRequest.sol` contract, or at least implements the same
+/// interface.
+fn extract_pb_bytes_from_solidity_file(path: &Path) -> Result<Vec<u8>, Error> {
+    let mut file = File::open(path).map_err(Error::SolidityFileCantOpen)?;
+    let mut contents = String::new();
+    file.read_to_string(&mut contents)
+        .map_err(Error::SolidityFileCantRead)?;
+
+    let hex_reg_ex: Regex =
+        Regex::new(r#"\s*constructor.*public\s*Request\s*\(\s*hex"([\da-f]+)"#)?;
```
Comment (tmpolaczyk) [inline toolkit/src/cli/data_requests.rs:55]: This regex does not match the example request from here: https://docs.witnet.io/tutorials/bitcoin-price-feed/compiling/

**Review by tmpolaczyk — APPROVED** [review]

File: toolkit/src/errors.rs [hunk 2]
```diff
@@ -7,7 +7,9 @@ use failure::Fail;
 #[derive(Debug, Fail)]
 #[fail(display = "error")]
 pub enum Error {
-    #[fail(display = "No bytes have been provided. Please use the --hex argument.")]
+    #[fail(
+        display = "No bytes have been provided. Please use the --hex or --from-solidity argument"
+    )]
     DataRequestNoBytes,
     #[fail(
         display = "The string provided in the --hex field is not a valid hexadecimal byte string: {}",
```
Comment (tmpolaczyk) [inline toolkit/src/errors.rs:15]: Now this error can happen without using the `--hex` field, when using the `from-solidity` option

File: toolkit/src/cli/data_requests.rs (hunk 1 above)
Comment (aesedepece) [reply toolkit/src/cli/data_requests.rs]: True. `public` is not used in constructors anymore. I adjusted the regex to account for either presence or absence of such modifier.

File: toolkit/src/errors.rs (hunk 2 above)
Comment (aesedepece) [reply toolkit/src/errors.rs:15]: This change was already covered in this PR, I don't know why GitHub didn't show it to you :shrug:

File: toolkit/src/cli/data_requests.rs (hunk 1 above)
Comment (tmpolaczyk) [reply toolkit/src/cli/data_requests.rs]: Now it does match the request but the bytes are not valid protobuf. So we should probably update that tutorial.

Comment (aesedepece) [reply toolkit/src/cli/data_requests.rs]: Yep, it seems those are totally outdated or otherwise simply broken.

**Review by tmpolaczyk — APPROVED** [review]

**Pull request merged**